server/api/auth_handler.py

```diff
@@ -42,22 +42,25 @@ def login():
         # TODO: log
         return fail_response(error_code.EMPTY_REQUIRED_FIELD), 400
     user_in_db = User.get_by_email(user_json["email"])
+
+    # To avoid time analysis side channel attack.
     if not user_in_db:
-        return fail_response(error_code.USER_NOT_EXIST), 400
-    user_in_db = user_in_db[0]
+        user_password_input = "$2b$12$0000000000000000000000000000000000lqwlNhRDOQcSSJOJwi"  # This Should always fail
```

Comment:
I would rather not.... this is not clean - instead just send the fail message here - the repeated code is easier to read than this

Reply:
Ok